### ISSUE TITLE
treesitter depends on nodejs

### ISSUE THREAD

**Issue by david16correa** [leogabac/leovim#1]
When opening a .tex file, the error shown in the image appears.
![PHOTO-2024-12-30-21-53-03](https://github.com/user-attachments/assets/ad6f21be-efdf-4ea5-8756-89f3fce78dad)
This is corrected when commenting
```lua
require("plugins.treesitter"),
```
in the init.lua file.

**Comment by leogabac** [comment]
uninstalling (Arch)
```
sudo pacman -Rns nodejs node-gyp nodejs-nopt npm semver yarn
```
and removinb `sudo rm -rf ~/.local/share/nvim`
then opening a tex file reproduces the problem

working on a fix

**Comment by leogabac** [comment]
update
when the treesitter CLI is not installed, treesitter specifically invokes for it only for latex files.
```
Error detected while processing BufReadPost Autocommands for "*":
Error executing lua callback: /usr/share/nvim/runtime/filetype.lua:35: Error executing lua: /usr/share/nvim/runtime/filetype.lua:36: BufReadPost Autocommands fo
r "*"..FileType Autocommands for "tex"..FileType Autocommands for "*": Vim(append):tree-sitter CLI not found: `tree-sitter` is not executable!
stack traceback:
        [C]: in function 'nvim_cmd'
        /usr/share/nvim/runtime/filetype.lua:36: in function </usr/share/nvim/runtime/filetype.lua:35>
        [C]: in function 'nvim_buf_call'
        /usr/share/nvim/runtime/filetype.lua:35: in function </usr/share/nvim/runtime/filetype.lua:10>
stack traceback:
        [C]: in function 'nvim_buf_call'
        /usr/share/nvim/runtime/filetype.lua:35: in function </usr/share/nvim/runtime/filetype.lua:10>
Press ENTER or type command to continue
```

latex is a particular parser that needs to be installed via the treesitter cli

**Comment by leogabac** [comment]
update
I disabled auto installation of parsers, this way treesitter will never try to install any type of parsers and they need to be installed manually.
```
:TSInstall <parser>
```
This way we avoid extra dependencies. Fixed by commit [052f9d5](https://github.com/leogabac/leovim/commit/052f9d57de7eb8fdca82d1252a5f1bda9ddf0d54)

For latex specifically, use the `tree-sitter-cli` and `:TSInstall latex`, it may require `nodejs`
